### PR TITLE
Gdpplotter 02

### PR DESCRIPTION
**Pull request by aldojasb**
Guys, I'm sending my version for the gdpplotter.

-I got problems with matplolib, and spend a lot of time learning that. It is harder compared to Altair. I could only do a very simple plot, but it's enough for this milestone.
-I updated the pyproject.toml (added numpy and matplolib as dependencies)
-I also wrote my test_gdpplotter. Apparently, it's working.

### CODE REVIEW

**Comment by gfairbro** [comment]
Hey @aldojasb great job this looks like a fair bit of coding. I am concerned that you are inputting a dataframe from file rather than just accepting a dataframe as input to your function. I am also not sure that this function is going to be robust enough to handle the different types of aggregators (other than just province and Canada-wide) that might be found in the data. I also have to make some changes to my function to deal with these differences, for example in this dataset where there is no location, it is "Special industry aggregations". (https://open.canada.ca/data/en/dataset/9f2116fc-265a-419f-a921-2d747c039f34) @RamiroMejia This is where i think we will need categories, or else a differen column name that generalizes better than Locations

**Comment by gfairbro** [comment]
Opened this issue to better capture this concern: https://github.com/UBC-MDS/gdphelper/issues/49

File: src/gdphelper/gdpplotter.py [hunk 1]
```diff
@@ -1,26 +1,84 @@
...
+    else:
+        raise ValueError(
+            'The aggragation should be "canada" or "province".'
+        )
+
+
+    #reading the database within the function
+    file = f"{data_name}.csv"
+    df = pd.read_csv(file)
+    print("database loaded")
```
Comment (GloriaWYY) [inline src/gdphelper/gdpplotter.py:43]: As Gabe mentioned, this function does not need to read data from files as this is done in upstream functions. 😸

File: src/gdphelper/gdpplotter.py [hunk 2]
```diff
@@ -1,26 +1,84 @@
...
+                                               fill_value=0, aggfunc=np.sum).reset_index()
+    #fixing columns
+    df_grouped_region_pivoted.columns = ['_'.join(col).strip() for col in df_grouped_region_pivoted.columns.values]
+    print("database was transformed")
+    
+    #selecting the province names
+    provinces = list(df_grouped_region_pivoted.columns)
+    provinces.remove('REF_DATE_')
+    provinces.remove('VALUE_Canada')
+    provinces
```
Comment (GloriaWYY) [inline src/gdphelper/gdpplotter.py:59]: I guess this is not needed for a function?

File: src/gdphelper/gdpplotter.py [hunk 3]
```diff
@@ -1,26 +1,84 @@
...
+        plt.show()
+
+    if aggregation == "canada":
+        #chart 02
+        df_grouped_region_pivoted.plot(x='REF_DATE_', y='VALUE_Canada')
+        plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)
+        plt.ylabel("GPD (CAD$ - millions)")
+        plt.xlabel("Years")
+        plt.title("GDP (CAD$ millions) VS Years", loc='center')
+        plt.show()
```
Comment (GloriaWYY) [inline src/gdphelper/gdpplotter.py:83]: I am a bit concerned about branch coverage for these parts - might just replace it with one `if`-`else` statement 😄

**Review by GloriaWYY — COMMENTED** [review]
Nice work! Same as what Gabe mentioned, this function takes a pandas DataFrame rather than read in a file. In the test function, would it be better to create a toy data frame like what Tiffany illustrated in today's lecture?

I also noticed there is a space in the test file (i.e. `test_  gdpplotter.py`) 😅

File: src/gdphelper/gdpplotter.py [hunk 4]
```diff
@@ -1,26 +1,84 @@
...
+        df_grouped_region_pivoted.plot(x='REF_DATE_', y=provinces)
+        plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)
+        plt.ylabel("GPD (CAD$ - millions)")
+        plt.xlabel("Years")
+        plt.title("GDP (CAD$ millions) VS Years", loc='center')
+        plt.show()
+
+    if aggregation == "canada":
+        #chart 02
+        df_grouped_region_pivoted.plot(x='REF_DATE_', y='VALUE_Canada')
```
Comment (RamiroMejia) [inline src/gdphelper/gdpplotter.py:78]: @gfairbro is this `REF_DATE` what you are concerned about?

File: src/gdphelper/gdpplotter.py [hunk 5]
```diff
@@ -1,26 +1,84 @@
...
+    #selecting the province names
+    provinces = list(df_grouped_region_pivoted.columns)
+    provinces.remove('REF_DATE_')
+    provinces.remove('VALUE_Canada')
+    provinces
+    if provinces == []:
+        print("WARNING: checking province names: there aren't provinces in this database")
+        print("therefore, use only option 'canada' for 'aggragation'.")
+    else:
+        print("checking province names: we have provinces in this dataset")
```
Comment (RamiroMejia) [inline src/gdphelper/gdpplotter.py:64]: might be a good idea to use try and Raise Error

File: src/gdphelper/gdpplotter.py [hunk 6]
```diff
@@ -1,26 +1,84 @@
...
+    
+    #drawing charst according to the selected option
```
Comment (RamiroMejia) [inline src/gdphelper/gdpplotter.py:66]:
```suggestion
    # drawing charts according to the selected option
```

File: src/gdphelper/gdpplotter.py [hunk 7]
```diff
@@ -1,26 +1,84 @@
 import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
 
-
-def gdpplotter(type_chart, type_feature):
+def gdpplotter(data_name, aggregation):
```
Comment (RamiroMejia) [inline src/gdphelper/gdpplotter.py:5]: @aldojasb so this is going to always retrieve plots by provice/country?

**Comment by aldojasb** [comment]
Thanks for the nice review, guys!! I'll fix all the points tonight. See you,

**Review by gfairbro — CHANGES_REQUESTED** [review]
Gonna try and see what happens when 'Request Changes' is selected here, to see if a new commit will trigger a new review!

**Pull request closed without merging**